fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -1106,7 +1106,9 @@ public void replayAddComputeNode(ComputeNode newComputeNode) {
         // update idToComputeNode
         newComputeNode.setBackendState(BackendState.using);
         idToComputeNodeRef.put(newComputeNode.getId(), newComputeNode);
-        tabletComputeNodeMapper.addComputeNode(newComputeNode.getId(), DEFAULT_RESOURCE_ISOLATION_GROUP_ID);
+        // TODO(cbrennan) Should this be using the given newComputeNode resource isolation group?
+        //  Technically the modify should get replayed right?
```

Comment:
Yes. Before, I was misunderstanding that sometimes the SystemInfoService is deserialized from an image instead of replaying all state-modifying transactions, and therefore replayAddComputeNode doesn't get called for all Compute nodes. That was when the compute nodes were in the state with the appropriate resource isolation group but not in the tablet compute node mapper.